src/main/java/com/pd/algorithms/BinarySearch.java

```diff
@@ -8,28 +8,28 @@ public class BinarySearch {
   public int size;
 
   public BinarySearch(int[] arr) {
+    Arrays.sort(arr);
     this.arr = arr;
     this.size = arr.length;
   }
 
   public boolean search(int num) {
-    Arrays.sort(arr);
     return binarySearch(arr, 0, size - 1, num);
   }
 
   public boolean binarySearch(int[] array, int low, int high, int value) {
```

Comment:
I guess we can make this method `private` since this is mainly to be called from the `search` public method. Could you please update this and also update the docs, if needed?